quipucords/api/inspectresult/model.py

```diff
@@ -83,7 +83,7 @@ class RawFact(models.Model):
     """A model of a raw fact."""
 
     name = models.CharField(max_length=1024)
-    value = models.TextField()
+    value = models.JSONField(null=True)
```

Comment:
I understand the OCP part is a WIP, but I'd try a less aggressive approach: customize the [JSONField encoder][1]

```python
from json import JSONEncoder
from pydantic import BaseModel
from datetime import datetime

class RawFactEncoder(JSONEncoder):
    def default(self, object):
        if isinstance(object, BaseModel):
            return object.dict()
        if isinstance(object, datetime):
            return object.isoformat()
        return super().default(object)
 ```

[1]: https://docs.djangoproject.com/en/4.1/ref/models/fields/#django.db.models.JSONField.encoder

Reply:
IoW, write the puppy above and then

```python
value = models.JSONField(null=True, encoder=RawFactEncoder)
```

Reply:
👍 Thanks, yes, this is much nicer and simpler doing that type of encoding for all RawFact values at the model, essentially adding support for OCP* types by default.